src/main/java/com/github/wovnio/wovnjava/Api.java

```diff
@@ -207,18 +221,44 @@ private URL getApiUrl(String lang, String body) throws UnsupportedEncodingExcept
         if (this.requestOptions.getCacheDisableMode() || this.requestOptions.getDebugMode()) {
             appendValue(sb, "&timestamp=");
             appendValue(sb, String.valueOf(System.currentTimeMillis()));
+        } else if (this.headers.isSearchEngineBot()) {
+          appendValue(sb, "&timestamp=");
+          appendValue(sb, getDynamicLoadingTimeStamp());
         }
         appendValue(sb, ")");
         return new URL(sb.toString());
     }
 
+    private String getDynamicLoadingTimeStamp() {
+      long secondsSinceEpoch = this.clock.millis() / DYNAMIC_LOADED_PAGE_TTL_MILI * DYNAMIC_LOADED_PAGE_TTL_MILI;
+      Date date = new Date(secondsSinceEpoch);
+      SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX");
+      format.setTimeZone(TimeZone.getTimeZone("JST"));
+      return format.format(date);
+    }
+
     private String hash(byte[] item) throws NoSuchAlgorithmException {
         MessageDigest md = MessageDigest.getInstance("MD5");
         md.update(item);
         byte[] digest = md.digest();
-        return DatatypeConverter.printHexBinary(digest).toUpperCase();
+        return this.encodeHexString(digest).toUpperCase();
     }
 
+    private String encodeHexString(byte[] byteArray) {
+      StringBuffer hexStringBuffer = new StringBuffer();
+      for (int i = 0; i < byteArray.length; i++) {
+          hexStringBuffer.append(byteToHex(byteArray[i]));
+      }
+      return hexStringBuffer.toString();
+    }
+
+    private String byteToHex(byte num) {
+      char[] hexDigits = new char[2];
+      hexDigits[0] = Character.forDigit((num >> 4) & 0xF, 16);
+      hexDigits[1] = Character.forDigit((num & 0xF), 16);
+      return new String(hexDigits);
+  }
```

Comment:
you could put this in a StringUtils class or something and write simple tests for it